### DISCUSSION TITLE
Detect Uno.Bootstrap.Wasm?

### DISCUSSION THREAD

**Discussion by SerratedSharp** [unoplatform/Uno.Wasm.Bootstrap#833] [Q&A] [answered]
Is there a recommended approach for detecting if the consuming app of my class library is using Uno.Wasm.Bootstrap from within C#?

My class library supports being consumed from either the new .NET 8 wasmbrowser template or Uno.Wasm.Bootstrap.  Because wasmbrowser's ImportAsync requires ES6 modules(embedded as RCL static web assets), and Uno uses AMD modules (embedded via /WasmScripts/ folder), the best I could work out is two version of my javascript declarations needed for the library's JSImports, but this impacts how JSImport is declared:

For Uno: `[JSImport("globalThis.MyStuff.GetItems")]`
For wasmbrowser: `[JSImport("MyStuff.GetItems", "MyStuff")]`

I have two sets of all my C# proxies with each version of JSIMport.  I would like to route calls to the appropriate version based on the presence of Uno.Bootstrap.Wasm. I tried checking for Uno related assemblies in CurrentDomain.GetAssemblies() but this seems like a bad approach.

P.S. I tried using JSHost,ImportAsync with Uno Bootstrap, but it seems like the RCL static web assets embedded in the class library don't resolve properly under Uno.  There's something in the dev server for the wasmbrowser template that allows it to resolve these requests during local development, even though the file doesn't physically exist in wwwroot anywhere at dev time.

**Comment by SerratedSharp** [comment]
I ended up adding a property to globalThis from my Uno specific JS loaded via WasmScripts `globalThis.IsFromUno = true;` and then using the JSHost.GlobalThis handle to check for it.  Not the most elegant solution but works.

```
internal static bool IsUnoWasmBootstrapLoaded => isUnoWasmBootstrapLoaded.Value;
private static Lazy<bool> isUnoWasmBootstrapLoaded = new Lazy<bool>(() =>
{
    bool isUnoPresent = false;
    if (JSHost.GlobalThis.HasProperty("IsFromUno"))
    {
        if (JSHost.GlobalThis.GetPropertyAsBoolean("IsFromUno"))
            isUnoPresent = true;
    }

    return isUnoPresent;
});

```

**Reply by jeromelaban** [reply, marked as answer]
You can use any of these environment variables to detect the bootstrapper: https://platform.uno/docs/articles/external/uno.wasm.bootstrap/doc/features-environment-variables.html
